src/components/Firestore/useAutoSelect.test.tsx

```diff
@@ -140,13 +149,13 @@ describe('useAutoSelect', () => {
           </Route>
         </Router>
       );
-      await wait();
+      await act(() => delay(100));
```

Comment:
curious if these delays waiting on promises are fixable with `await act(() => jest.runAllTimers())` or similar?

else, LGTM!

Reply:
`useAutoSelect` does not wait on promises, instead the problem is that it triggers `useEffect` which triggers `<Redirect>` which in turn re-renders. So I don't think `jest.runAllTimers()` can help here. Also there are many test cases that cover cases that should not trigger redirects -- just calling `jest.runAllTimers()` is not a great proof that it won't redirect in the next frame. I think the way to go is to mock out `<Redirect>` so it renders some text instead of side-effects, and I'm happy to review a PR for that.

For other promise-based stuff I've switched most to use Deferred. Deferred is the most reliable and controllable way to test those cases and we should prefer that to `jest.runAllImmediates()` if possible. (If a component uses Promise in a way that makes using Deferred hard, we should consider refactoring first, or turn fake timers on and call `jest.runAllImmediates()` just for a few tests.)